File: lottie-preview-plugin/src/main/java/com/lottiepreview/plugin/toolwindow/LottiePreviewWindowFactory.java
package com.lottiepreview.plugin.toolwindow;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.wm.ToolWindow;
import com.intellij.openapi.wm.ToolWindowFactory;
import com.intellij.ui.content.Content;
import org.jetbrains.annotations.NotNull;

public class LottiePreviewWindowFactory implements ToolWindowFactory {
    @Override
    public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindow toolWindow) {
        LottiePreviewPanel panel = new LottiePreviewPanel(project);
        Content content = toolWindow.getContentManager().getFactory().createContent(panel, "", false);
        toolWindow.getContentManager().addContent(content);
    }
}

File: lottie-preview-plugin/src/main/kotlin/com/lottiepreview/plugin/toolwindow/LottiePreviewWindowFactory.kt
package com.lottiepreview.plugin.toolwindow

import com.intellij.openapi.project.Project
import com.intellij.openapi.wm.ToolWindow
import com.intellij.openapi.wm.ToolWindowFactory

class LottiePreviewWindowFactory : ToolWindowFactory {
    override fun createToolWindowContent(project: Project, toolWindow: ToolWindow) {
        val panel = LottiePreviewPanel()
        val content = toolWindow.contentManager.factory.createContent(panel, "", false)
        toolWindow.contentManager.addContent(content)
    }
}
